perun-cli/Perun/beans/HashedGenData.pm

```diff
@@ -0,0 +1,304 @@
+package Perun::beans::HashedGenData;
+
+use strict;
+use warnings;
+
+use Perun::Common;
+use Data::Dumper;
+
+sub new
+{
+	bless({});
+}
+
+sub fromHash
+{
+	my $class = shift;
+	$class = ref $class if ref $class;
+
+	my $self = { };
+	my $attributes = { };
+	my $hierarchy = { };
+	my $hash;
+
+	if ((@_ == 1) && (ref($_[0]) eq 'HASH')) {
+		$hash = $_[0];
+	} else {
+		my %hash = @_;
+		$hash = \%hash;
+	}
+
+	$self->{_hierarchy} = $hash->{'hierarchy'};
```

Comment:
One more small thing. When you chose not to quote hash keys, it would be better to be consistent with all of them.

Reply:
Removed.